clientv3/concurrency/election.go

```diff
@@ -39,17 +39,14 @@ type Election struct {
 }
 
 // NewElection returns a new election on a given key prefix.
-func NewElection(client *v3.Client, pfx string) *Election {
+func NewElection(client *ClientWithSession, pfx string) *Election {
```

Comment:
@xiang90 Just `NewElection(s *Session, pfx string)`. The `Session` can have a function `func (s *Session) Client() *clientv3.Client { return s.client}` for anything that needs to access etcd through the session. The `ClientWithSession` struct probably doesn't need to be part of the concurrency package.